### PR TITLE
Display MantisGraph pages under their own menu

### PR DESCRIPTION
**Pull request by dregad**
Fixes [#26139](https://mantisbt.org/bugs/view.php?id=26139)

### CODE REVIEW

**Comment by dregad** [comment]
Without feedback, I assume everyone's OK with this and will proceed with merging this week-end.

**Comment by atrol** [comment]
Tested a bit, looks good.
Hope to be able to invest some time also in code review.

I am not sure if it's a good idea to change non-english language files, as I don't know how Translatewiki works.

You missed to remove `$s_plugin_MantisGraph_synthesis_link = 'Sintezo';` from `strings_eo.txt`

File: core/html_api.php [hunk 1]
```diff
@@ -564,35 +564,112 @@ function print_subproject_menu_bar( $p_current_project_id, $p_parent_project_id,
...
-					$t_menu_options[] = $t_callback_menu_options;
-				}
+				# Cooked link
+				echo $t_item;
 			}
 		}
+
+		# Plugins menu items - these are html hyperlinks (<a> tags)
+		foreach( plugin_menu_items( $p_event ) as $t_item ) {
+			$t_active = $p_current_page && strpos( $t_item, $p_current_page ) !== false
```
Comment (atrol) [inline core/html_api.php]: `$t_active` is not used

Comment (dregad) [reply core/html_api.php]: Fixed

**Comment by dregad** [comment]
> I am not sure if it's a good idea to change non-english language files, as I don't know how Translatewiki works.

@mantisbt/translatewiki-net can you please confirm if this is OK or not ?

> You missed to remove $s_plugin_MantisGraph_synthesis_link = 'Sintezo'; from strings_eo.txt

Fixed

**Comment by dregad** [comment]
@translatewiki your feedback would be appreciated.

**Comment by dregad** [comment]
> I am not sure if it's a good idea to change non-english language files, as I don't know how Translatewiki works.

See [Discussion on translatewiki.net support channel](https://translatewiki.net/wiki/Support#Deleting_unused_strings_from_translated_language_files_56747
).

**Comment by atrol** [comment]
Translatewiki wrote

> If you delete the English strings (the source strings), the next export will take care of deleting the translations as well. 

Seems not to be true.
E.g. when running admin/test_langs.php you get warnings like
```
Testing language file 'strings_afrikaans.txt' (phase 1)...
WARNING: '$s_plugin_MantisGraph_enabled' is not defined in the English language file
WARNING: '$s_plugin_MantisGraph_disabled' is not defined in the English language file
```

**Comment by siebrand** [comment]
| If you delete the English strings (the source strings), the next export will take care of deleting the translations as well.

It is true, and there is a but :). But we only export language files when changes have been made to it recently. So if no one makes changes to Dutch for example, the keys that no longer exist in the English file, will not be cleaned from the Dutch file, simply because we're not (re-)generating it.

My advise is to no worry about superfluous entries in non-English language files if they are not in the way of a correctly functioning application.

**Comment by dregad** [comment]
Thanks for the feedback @siebrand. So if I understand correctly, you recommend not to edit foreign language files (but it does not hurt if we do).

Since you're here, there is still one question that needs answering from @translatewiki (as [posted on translatewiki.net support forum](https://translatewiki.net/wiki/Thread:Support/Deleting_unused_strings_from_translated_language_files/reply_(2))): 

> [renaming a language string](https://github.com/mantisbt/mantisbt/pull/1564/commits/0296fd0f782e7fa258e323471890fba4f1ca33f0) (i.e. not the translation itself, but the string's identifier) ==> this is the one I'm really not sure about. I would think yes based on what the FAQ says, but since it also mentions informing the staff...

**Comment by Abijeet** [comment]
Hi @dregad 

Answered on the support ticket, but here's the summary,

> _renaming a language string_ --> This is also fine. Thanks for the ping regarding this. We will review and do the necessary renaming on our end, which is quite easy.

Regards,

**Comment by dregad** [comment]
@Abijeet great, thanks for the feedback !

**Pull request merged**